metricproviders/newrelic/newrelic.go

```diff
@@ -41,13 +66,30 @@ type NewRelicClient struct {
 }
 
 // Query executes a NRQL query against the given New Relic account
-func (n *NewRelicClient) Query(query string) ([]nrdb.NRDBResult, error) {
-	results, err := n.Nrdb.Query(n.AccountID, nrdb.NRQL(query))
```

Comment:
There's an equivalent [function](https://github.com/newrelic/newrelic-client-go/blob/36d7acf42b8bd408e77dd1898ab148ee43922a15/pkg/nrdb/nrdb_api.go#L107) that supports passing timeout in the `nrdb` package but it [pulls way more data](https://github.com/newrelic/newrelic-client-go/blob/36d7acf42b8bd408e77dd1898ab148ee43922a15/pkg/nrdb/nrdb_api.go#L213) than the current function which by itself can impact latency.
That's why I switched to the NerdGraph package to send the timeout in the query without incurring in additional data over the wire. I actually slimmed down the query compared to the [used](https://github.com/newrelic/newrelic-client-go/blob/36d7acf42b8bd408e77dd1898ab148ee43922a15/pkg/nrdb/nrdb_api.go#L40) in the `Query` function to just what we need for this metric provider so the GQL API has to do less work to respond (and even less data transmitted back)